instance/models/openedx_instance.py

```diff
@@ -172,20 +205,38 @@ def get_active_appservers(self):
             return self.ref._cached_active_appservers
         return self.appserver_set.filter(_is_active=True)
 
+    def domain_nginx_regex(self, site_name):
+        """
+        Regex that matches either the internal or external URL for the site.
+
+        This is meant exclusively for filling in the server_name regex in nginx configs.
+        """
+        domain_attrs = ['external_{}_domain', 'internal_{}_domain']
+        domains = [getattr(self, attr.format(site_name), None) for attr in domain_attrs]
+        domains = [domain for domain in domains if domain is not None]
+        choices = '|'.join(map(re.escape, domains))  # pylint: disable=bad-builtin
+        return '~^({})$'.format(choices)
```

Comment:
@e-kolpakov This is just copied from the existing code, and I introduced it originally.  In nginx, `~` marks a regular expression.  This is why I called the function `..._nginx_regex`.  We could add the `~` in the template instead of here, if you prefer.  I don't feel strongly either way.